src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts

```diff
@@ -64,13 +67,59 @@ export class ItsLearningPersonsEventHandler {
     public constructor(
         private readonly logger: ClassLogger,
         private readonly itsLearningService: ItsLearningIMSESService,
+        private readonly personRepo: PersonRepository,
         configService: ConfigService<ServerConfig>,
     ) {
         const itsLearningConfig: ItsLearningConfig = configService.getOrThrow<ItsLearningConfig>('ITSLEARNING');
 
         this.ENABLED = itsLearningConfig.ENABLED === 'true';
     }
 
+    @EventHandler(PersonRenamedEvent)
+    public async personRenamedEventHandler(event: PersonRenamedEvent): Promise<void> {
+        this.logger.info(`Received PersonRenamedEvent, ${event.personId}`);
+
+        if (!this.ENABLED) {
+            return this.logger.info('Not enabled, ignoring event.');
+        }
+
+        return this.mutex.runExclusive(async () => {
+            const person: Option<Person<true>> = await this.personRepo.findById(event.personId);
```

Comment:
Will it be okay if the PersonRenamedEvent object also delivers the Person Data (Vorname, Nachname & Referrer) that you need so that you do not have to call the PersonRepository again?

Reply:
Added more fields to the event